.github/workflows/on-prerelease.yml

```diff
@@ -0,0 +1,237 @@
+  MANIFEST_FILE: "_manifest/spdx_2.2/manifest.spdx.json"
+  PACKAGE_NAME: "EdFi.Dms.Api"
+  REF: ${{ github.ref_name }}
+
+jobs:
+  pack:
+    name: Build and Pack
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: pwsh
+    outputs:
+      hash-code: ${{ steps.hash-code.outputs.hash-code }}
+      dms-api-version: ${{ steps.versions.outputs.dms-api }}
+
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          fetch-depth: 0
+
+      - name: Set Version Numbers
+        id: versions
+        run: |
+          $apiPrefix = "v"
+
+          # Install the MinVer CLI tool
+          &dotnet tool install --global minver-cli
+
+          $apiVersion = $(minver -t $apiPrefix)
+
+          # Full release name
+          "dms-api=$apiVersion" >> $env:GITHUB_OUTPUT
+
+          # SemVer
+          "dms-api-semver=$($apiVersion -Replace $apiPrefix)" >> $env:GITHUB_OUTPUT
+
+      - name: Publish .NET Assemblies
+        run: |
+          $apiVersion = "${{ steps.versions.outputs.dms-api-semver }}"
+
+          ./build.ps1 -Command BuildAndPublish `
+              -Configuration Release `
+              -APIVersion $apiVersion
+
+      - name: Setup Nuget.exe
+        uses: nuget/setup-nuget@fd9fffd6ca4541cf4152a9565835ca1a88a6eb37 #v1.1.1
+
+      - name: Create NuGet Packages
+        if: success()
+        run: |
+          $apiVersion = "${{ steps.versions.outputs.dms-api-semver }}"
+
+          "Building ${{ env.PACKAGE_NAME }} NuGet Package" | Write-Output
+          # dms API
+          ./build.ps1 -Command Package `
+              -APIVersion  $apiVersion `
+              -Configuration Release
+
+      - name: Generate hash for NuGet package
+        id: hash-code
+        shell: bash
+        run: |
+          echo "hash-code=$(sha256sum *.nupkg | base64 -w0)" >> $GITHUB_OUTPUT
+
+      - name: Upload Packages as Artifacts
+        if: success()
+        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 # v4.3.0
+        with:
+          name: "${{ env.PACKAGE_NAME }}-NuGet"
+          path: ./*.nupkg
+          if-no-files-found: error
+          retention-days: 30
+
+  sbom-create:
+    name: Create SBOM
+    runs-on: ubuntu-latest
+    needs: pack
+    permissions:
+      actions: read
+      contents: write
+    outputs:
+      sbom-hash-code: ${{ steps.sbom-hash-code.outputs.sbom-hash-code }}
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Get Artifacts
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a #v3.0.2
+        with:
+          name: ${{ env.PACKAGE_NAME }}-NuGet
+
+      - name: Generate Software Bill of Materials (SBOM) - API
+        shell: pwsh
+        run: |
+          $packageName = "${{ env.PACKAGE_NAME }}"
+
+          $url = "https://github.com/microsoft/sbom-tool/releases/latest/download/sbom-tool-linux-x64"
```

Comment:
This suggestion is a significant enough change that we can consider applying it in a separate ticket, since this ticket is blocking everything else. Please use your discretion.